src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -175,13 +179,12 @@ def delete(self, request, organization):
 
         if serializer.is_valid():
             data = serializer.validated_data
-            base_filter = {
-                "organization": organization,
-                "project": project,
-                "transaction": data["transaction"],
-            }
 
-            TeamKeyTransaction.objects.filter(**base_filter, team__in=data["team"]).delete()
+            TeamKeyTransaction.objects.filter(
+                organization=organization,
+                project_team__in=ProjectTeam.objects.filter(project=project, team__in=data["team"]),
```

Comment:
When a team is removed from a project do you also need to update the team key transactions?

Reply:
The `TeamKeyTransaction` model has a cascading delete that should handle this case.